DOC: Note about collapsing multi-dimensional coordinates

dkillick opened the pull request:
Adds a note on partially collapsing multi-dimensional coordinates to the userguide &sect;9.1, `iris.cube.Cube.collapsed` and `iris.coords.Coord.collapsed`.

pelson commented on docs/iris/src/userguide/cube_statistics.rst:
```diff
@@ -98,6 +98,22 @@ in the gallery takes a zonal mean of an ``XYT`` cube by using the
+
+    Instead you must either fully collapse the coordinate (in this case
+    requiring that both ``grid_latitude`` or ``grid_longitude`` coordinates be
+    collapsed together in a single collapse operation) or not collapse the
```
Third option - remove the surface altitude.

pelson commented on docs/iris/src/userguide/cube_statistics.rst:
```diff
@@ -98,6 +98,22 @@ in the gallery takes a zonal mean of an ``XYT`` cube by using the
+    collapsed together in a single collapse operation) or not collapse the
+    coordinate at all.
+
+    Multi-dimensional derived coordinates (such as ``altitude`` in the above
```
Perhaps too much detail for the userguide.

pelson commented:
:+1: in principle. My major hesitation is on the addition to the userguide. Could you take a look at the built userguide, and tell me if you think the addition makes the flow work as well in the context of it being a document designed to be used as a readable narrative? The addition to the reference documentation I am absolutely :+1: on.

pelson commented on docs/iris/src/userguide/cube_statistics.rst:
```diff
@@ -98,6 +98,22 @@ in the gallery takes a zonal mean of an ``XYT`` cube by using the
 
 
+.. note::
+    You cannot partially collapse a multi-dimensional coordinate. Thus in the
```
Action: make into a single sentence and cross reference "reference docs".

pelson commented on lib/iris/coords.py:
```diff
@@ -943,11 +943,21 @@ def cell(self, index):
 
         Replaces the points & bounds with a simple bounded region.
 
+        .. note::
```
I'm not sure this note is well targeted for coordinates.

pelson commented on lib/iris/cube.py:
```diff
@@ -2945,6 +2945,16 @@ def collapsed(self, coords, aggregator, **kwargs):
                 cube.collapsed(['latitude', 'longitude'],
                                iris.analysis.VARIANCE)
 
+        .. note::
```
:+1: for more detail in here which we can cross reference.

dkillick commented:
As per @pelson's suggestions above, I have:
 * simplified the addition to the userguide by removing the `note` directive and almost all of the information,
 * placed the information removed from the userguide into the `note` directive in the docstring for `iris.cube.Cube.collapsed` and added a `ref` directive for this `note`, and
 * added a link to this `note` to the userguide and the docstring for `iris.coords.Coord.collapsed`, which I have also simplified in favour of this link.
